pkgs/native_assets_builder/test_data/treeshaking_native_assets/hook/build.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+const packageName = 'treeshaking_native_assets';
+
+void main(List<String> arguments) async {
+  await build(arguments, (config, output) async {
+    final cbuilder = CBuilder.library(
+      name: packageName,
+      assetName: 'staticlib',
+      sources: [
+        'src/native_add.c',
+        'src/native_multiply.c',
+      ],
+      dartBuildFiles: ['hook/build.dart'],
+      linkModePreference: LinkModePreference.static,
+    );
+    await cbuilder.run(
+      buildConfig: config,
+      buildOutput: output,
+      logger: Logger('')
+        ..level = Level.ALL
+        ..onRecord.listen((record) {
+          print('${record.level.name}: ${record.time}: ${record.message}');
+        }),
+      linkInPackage: packageName,
+    );
+    _placeholderAsset(config, output);
+  });
+}
+
+void _placeholderAsset(BuildConfig config, BuildOutput output) {
```

Comment:
Please add a TODO with a reference to https://github.com/dart-lang/native/issues/1084